src/include/OSL/platform.h

```diff
@@ -60,27 +61,45 @@
 // Define OSL_APPLE_CLANG_VERSION to hold an encoded Apple Clang version
 // (e.g. 70002 for clang 7.0.2), or 0 if not an Apple Clang release.
 #if defined(__clang__) && defined(__apple_build_version__)
+#  if defined(__INTEL_LLVM_COMPILER)
+#    error Not expected for __INTEL_LLVM_COMPILER to be defined with an __apple_build_version__
+#  endif
 #  define OSL_APPLE_CLANG_VERSION (10000*__clang_major__ + 100*__clang_minor__ + __clang_patchlevel__)
 #else
 #  define OSL_APPLE_CLANG_VERSION 0
 #endif
+// The classic Intel(r) C++ Compiler on OSX may still define __clang__
+// combine with testing OSL_INTEL_COMPILER to further differentiate if needed.
 
-// Define OSL_INTEL_COMPILER to hold an encoded Intel compiler version
-// (e.g. 1900), or 0 if not an Intel compiler.
+// Define OSL_INTEL_COMPILER to hold an encoded Intel(r) C++ Compiler version
+// (e.g. 1900), or 0 if not an Intel(r) C++ Compiler.
 #if defined(__INTEL_COMPILER)
 #  define OSL_INTEL_COMPILER __INTEL_COMPILER
 #else
 #  define OSL_INTEL_COMPILER 0
 #endif
 
-// Intel's compiler on OSX may still define __clang__
-// and we have need to know when using a true clang compiler
-#if !defined(__INTEL_COMPILER) && defined(__clang__)
-    #define OSL_NON_INTEL_CLANG  __clang__
+// Define OSL_INTEL_LLVM_COMPILER to hold an encoded Intel(r) LLVM Compiler version
+// (e.g. 20220000), or 0 if not an Intel(r) LLVM Compiler.
+// Define OSL_INTEL_CLANG_VERSION to hold the encoded Clang version the
+// Intel(r) LLVM Compiler is based on (e.g. 140000),
+// or 0 if not an Intel(r) LLVM compiler.
+#if defined(__INTEL_LLVM_COMPILER)
+#  define OSL_INTEL_LLVM_COMPILER __INTEL_LLVM_COMPILER
+#  define OSL_INTEL_CLANG_VERSION (10000*__clang_major__ + 100*__clang_minor__ + __clang_patchlevel__)
```

Comment:
Are both of these necessary? For the other compilers, we only define OSL_COMPILERNAME_VERSION, set to some encoding of the version, and to 0 if that is not the compiler being used. If `#if OSL_INTEL_LLVM_COMPILER` isn't every going to evaluate differently than `#if OSL_INTEL_CLANG_VERSION`, then let's just eliminate the former one. (I don't have a preference for whether you call it OSL_INTEL_CLANG_VERSION or OSL_INTEL_LLVM_VERSION... pick whichever you think is better, and I'll change the OIIO side to conform.)

Reply:
Depends on what you want to check on.  If you have a workaround that is more clang version based, then it might be nice to check the clang version, instead of having to figure out which __INTEL_LLVM_COMPILER version's map to which clang versions (20220000 __INTEL_LLVM_COMPILER  maps to clang version 140000.  If only 1, then OSL_INTEL_LLVM_COMPILER as that might rev several times without changing the underlying clang compiler.

Reply:
Oh, I see what you mean. Are you saying that `OSL_INTEL_LLVM_COMPILER` holds the actual Intel version number, while `OSL_INTEL_CLANG_VERSION` holds the version of the clang that it's emulating or derived from?

I misunderstood, and thought you were using OSL_INTEL_LLVM_COMPILER simply as a bool of sorts.

So, in that case, let me change my recommendation. Let's change  `OSL_INTEL_LLVM_COMPILER` to `OSL_INTEL_LLVM_COMPILER_VERSION` to make clear that it's a version number, not just a bool, and thus match `OSL_GNUC_VERSION`, `OSL_APPLE_CLANG_VERSION`, etc.

Reply:
Seems valid reasoning.  This means the older 
OSL_INTEL_COMPILER
which also has the compiler version (its not just a bool), should be renamed as well.  And you were already suggesting
OSL_INTEL_CLASSIC_COMPILER, so should it be 
OSL_INTEL_CLASSIC_COMPILER_VERSION
?